csm_web/logistinator/models.py

```diff
@@ -1,3 +1,108 @@
+import datetime
 from django.db import models
+from django.conf import settings
+from django.utils import timezone
 
-# Create your models here.
+
+class Availability(models.Model):
+    """
+    Byte order is little endian, but the days are big endian.
+
+    000 ... 111
+    MON ... SUN
+
+    Within a day, it is also little endian
+
+    8AM 9AM ... 12AM
+    000 111 ... 111
+    """
+
+    # TODO Is there a cleaner way of doing this?
+    MON = 0
+    TUE = 1
+    WED = 2
+    THUR = 3
+    FRI = 4
+    SAT = 5
+    SUN = 6
+
+    BYTE_ORDER = "little"
+
+    DAYS = 7
+    HOURS_PER_DAY = 16
+    INTERVALS_PER_HOUR = 4
+    MINS_PER_INTERVAL = 60 // INTERVALS_PER_HOUR
+
+    INTERVAL = datetime.timedelta(minutes=MINS_PER_INTERVAL)
+    START_TIME = datetime.time(7, tzinfo=timezone.get_current_timezone())  # 7AM
+    END_TIME = datetime.time(23, tzinfo=timezone.get_current_timezone())  # 11PM
+
+    INTERVAL_COUNT = INTERVALS_PER_HOUR * HOURS_PER_DAY * DAYS
+
+    # By default, all slots are available
+    default_bitstring = (1 << INTERVAL_COUNT) - 1
```

Comment:
Think it would be safer to make everything unavailable by default instead...

Reply:
ok